### PR TITLE
das: add extra debug logs for sampled headers

### PR DESCRIPTION
**Pull request by walldiss**
Adds some logs for easier debugging
Resolves https://github.com/celestiaorg/celestia-node/issues/1155

### CODE REVIEW

**Review by Bidon15 — DISMISSED** [review]
Smooth loggerator 
Thanks for the work 🙏

File: das/daser.go [hunk 1]
```diff
@@ -168,6 +169,8 @@ func (d *DASer) sample(ctx context.Context, h *header.ExtendedHeader) error {
 			"square width", len(h.DAH.RowsRoots), "data root", h.DAH.Hash(), "err", err)
 		return err
 	}
+	log.Debugw("sampled header", "height", h.Height, "hash", h.Hash(),
+		"square width", len(h.DAH.RowsRoots), "data root", h.DAH.Hash(), "finished (s)", time.Since(start))
```
Comment (Wondertan) [inline das/daser.go:173]: We might want to move it to the worker loop because metrics introduce the same time tracking, and doing it twice is wasteful.

Comment (walldiss) [reply das/daser.go]: Good point. updated

File: das/worker.go [hunk 2]
```diff
@@ -43,10 +43,18 @@ func (w *worker) run(
...
 		h, err := getter.GetByHeight(ctx, curr)
 		if err == nil {
-			err = sample(ctx, h)
+			log.Debugw("got header from header store", "height", h.Height, "hash", h.Hash(),
+				"square width", len(h.DAH.RowsRoots), "data root", h.DAH.Hash(), "finished (s)", time.Since(startGet))
+
+			startSample := time.Now()
+			if err = sample(ctx, h); err == nil {
+				log.Debugw("sampled header", "height", h.Height, "hash", h.Hash(),
+					"square width", len(h.DAH.RowsRoots), "data root", h.DAH.Hash(), "finished (s)", time.Since(startSample))
```
Comment (Wondertan) [inline das/worker.go:56]: One last thing. Let's make it more readable by following Uber's Go guidelines and [reduce nesting](https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting).

File: das/daser_test.go [hunk 3]
```diff
@@ -297,6 +299,7 @@ func (m getterStub) Head(context.Context) (*header.ExtendedHeader, error) {
 
 func (m getterStub) GetByHeight(_ context.Context, height uint64) (*header.ExtendedHeader, error) {
 	return &header.ExtendedHeader{
+		Commit:    &types.Commit{},
```
Comment (renaynay) [inline das/daser_test.go:302]: What was the issue that this solves?

File: das/worker.go (hunk 2 above)
Comment (renaynay) [reply das/worker.go:56]: +1

File: das/worker.go [hunk 4]
```diff
@@ -43,17 +43,30 @@ func (w *worker) run(
...
+			"square width", len(h.DAH.RowsRoots), "data root", h.DAH.Hash(), "finished (s)", time.Since(startGet))
 
+		startSample := time.Now()
+		err = sample(ctx, h)
 		if errors.Is(err, context.Canceled) {
 			// sampling worker will resume upon restart
 			break
 		}
 		w.setResult(curr, err)
+
```
Comment (Wondertan) [inline das/worker.go:67]: I assume one `setResult` with nil is always required, or should we also check `if err != nil` in here?

**Review by Wondertan — APPROVED** [review]

**Review by Bidon15 — DISMISSED** [review]

File: das/worker.go (hunk 4 above)
Comment (walldiss) [reply das/worker.go:67]: No need, error should be propagated into setResults.

**Review by renaynay — DISMISSED** [review]

**Review by Bidon15 — APPROVED** [review]
🚢 🚢 🚢 🚢 🚢

**Pull request merged**